src/Atmos/Model/moisture.jl

```diff
@@ -15,27 +15,40 @@ end
 function gradvariables!(::MoistureModel, transform::Vars, state::Vars, aux::Vars, t::Real)
 end
 
-# TODO: rewrite to use MoistThermo
-struct DryModel <: Atmos.MoistureModel
+function internal_energy(m::MoistureModel, state::Vars, aux::Vars)
+  T = eltype(state)
+  q_pt = get_phase_partition(m, state)
+  ρinv = 1 / state.ρ
+  ρe_kin = ρinv*sum(abs2, state.ρu)/2
+  ρe_pot = state.ρ * grav * aux.coord.z
+  ρe_int = state.ρe - ρe_kin
+  # ρe_int = state.ρe - ρe_kin - ρe_pot # FIXME: Should we always include/exclude ρe_pot?
```

Comment:
Specific internal energy should always be `ρe_int = state.ρe - ρe_kin - ρe_pot`.

Reply:
I'll update this